src/commands/duckrelated.py

```diff
@@ -0,0 +1,110 @@
+import os
+import random
+import aiohttp
+from discord import app_commands, Interaction, Embed
+from discord.ext import commands
+from dotenv import load_dotenv
+from typing import Optional, List
+
+from constants.duck_data import DUCK_FACTS, DUCK_JOKES
+
+load_dotenv()
+
+TENOR_API_KEY = os.getenv("TENOR_API_KEY")
+TENOR_API_URL = "https://g.tenor.com/v2/search"
+DUCK_PIC_API_URL = (
+    "https://random-d.uk/api/v2/random?type=jpg"  # Duck picture API by random-d.uk
+)
+
+
+class DuckCommands(app_commands.Group):
+    def __init__(self):
+        super().__init__(name="duck", description="Fun duck-related commands")
+
+    async def get_tenor_gif(self, search_term: str = "duck") -> Optional[str]:
+        """Fetch a random GIF from Tenor API."""
```

Comment:
This function might be better suited to being a utility function that's independent from the `DuckCommands` class

Reply:
Yeah that's fair, we might use tenor GIF's for commands that use the Google Gemini API like #30.